### PR TITLE
remove other notification actions

### PR DESCRIPTION
**Pull request by murilommen**
details:
- it has been quite a while since we only support `GlobalActions`
- this PR removes both the helper methods to turn models into GlobalActions as well as typing references to other models
- it might break existing customers using the tuned models, so the next release should communicate this properly
- this PR also fixes a bunch of typing issues after running pyright on the repo

### CODE REVIEW

File: whylabs_toolkit/monitor/manager/monitor_setup.py [hunk 1]
```diff
@@ -309,7 +308,7 @@ def __set_dataset_matrix_for_dataset_metric(self) -> None:
                 logger.warning(
                     "ColumnMatrix is not configurable with a DatasetMetric." "Changing it to DatasetMatrix instead"
                 )
-                self._target_matrix = DatasetMatrix(segments=self._target_matrix.segments)
+                self._target_matrix = DatasetMatrix(type=TargetLevel.dataset, segments=self._target_matrix.segments)
```
Comment (christinedraper) [inline whylabs_toolkit/monitor/manager/monitor_setup.py:309]: why do you need to specify type in this case?

Comment (murilommen) [reply whylabs_toolkit/monitor/manager/monitor_setup.py:309]: just to address pyright's complaints, type is still auto-defined for users

File: whylabs_toolkit/monitor/models/monitor.py [hunk 2]
```diff
@@ -217,7 +180,7 @@ class Monitor(NoExtrasBaseModel):
         regex="[0-9a-zA-Z \\-_]+",
     )
     tags: Optional[  # type: ignore
-        List[constr(min_length=3, max_length=32, regex="[0-9a-zA-Z\\-_]")]  # noqa F722
+        List[constr(min_length=3, max_length=32, regex="[0-9a-zA-Z\\-_]")]  # type:ignore # noqa: F722
```
Comment (christinedraper) [inline whylabs_toolkit/monitor/models/monitor.py]: You can probably fix the type error here and in analyzer by changing constr to Annotated like I did with monitor schema

Comment (murilommen) [reply whylabs_toolkit/monitor/models/monitor.py]: just caught an error by defining `analyzerIds` to be a `str` (same as you did for the monitor schema), so I believe we should also update that there!

Comment (christinedraper) [reply whylabs_toolkit/monitor/models/monitor.py]: Did you try changing the `List[constr...` on tags?   So maybe `type:ignore` can go away

**Review by christinedraper — APPROVED** [review]

**Pull request merged**